Werkgroep API design rules/naamgeving.md

```diff
@@ -0,0 +1,56 @@
+# Naamgeving
+## Generieke naamconventies
+### Speciale tekens
+- Diakrieten normaliseren naar basis-karakter
+- Symbolen/leestekens (met uitzondering van een koppelteken) weglaten
+ 
+## Resource namen en path segments
+Omdat resources dingen beschrijven (in plaats van acties), worden resource namen als zelfstandig naamwoorden beschreven. Enkelvoudige resources worden in enkelvoud geschreven, collection resources in meervoud:
+(etc--kan voortborduren op 3.1 Resources van de API Design Rules).
+
+In het geval van samengestelde resource namen of path segments, wordt de kebab-case casing conventie gehanteerd.
+
+### Voorbeelden:
+- `/gebouw`
+- `/vergunningen`
+- `/organisatie-codes`
+ 
+### Niet doen:
+- Gebruik geen operaties zoals gebruikelijk is bij RPC-style API's (bv /getResource).
+- Refereer niet in de naam dat iets een API is (bv /resource_api).
+- Gebruik geen bestands extensies in de resource namen (bv /resource.json).
+    - Het gebruik van Content Negotiation via de Accept header heeft de voorkeur, al zijn API aanbieders vrij om convenience endpoints aan te bieden op basis van extensies.
+- Voorkom waar mogelijk het gebruik van afkortingen voor concepten, zoals `/coords` ipv `/coordinaten`. Een uitzondering hierop is het gebruik van breed bekende afkortingen als kwalificatie in veldnamen zoals `bagNummeraanduiding`. 
+- Gebruik geen samengestelde namen voor geneste objecten (bv /adres_straat).
+    - Gebruik hiervoor een sub-object 'adres' met daarin een 'straat' parameter.
+
+## Resource bodies
+- Gebruik JSON of JSON gebaseerde documenttypen (bv JSON+HAL)
+- Gebruik camelCase elementen
+
+## Request parameters
+Het voorstel voor request parameters (query parameters en in de request body) is:
```

Comment:
ik zie hier nu geen voorstel voor casing. Moet dat niet zijn camelCase? Zoveel mogelijk moet een query parameter gelijk zijn aan de resource property waar het op filtert.

> query parameters en in de request body
En voor headers? En path parameters? Path parameter regels moeten gelijk zijn aan de manier waarop properties en query parameters worden geschreven, want een property kan een queryparameter zijn op een endpoint, en een path parameter in een ander endpoint.

Reply:
@fsamwel wij zijn uitgekomen op spinal-case voor path segmenten en query parameters, omdat niet alle server producten en frameworks even goed omgaan met case sensitive URI's. Het klopt dat dit afwijkt van de gegevens in de JSON body en dat je bij filteren in de query een andere casing gebruikt dan de naam in het JSON object. Dus als je een veld "gebouwType" in de JSON body hebt en daarop wilt filteren, gebruik je iets van "?gebouw_type=...". Wij hebben dit binnen de werkgroep besproken en daarbij is besloten dat dit geen probleem is. Kun jij je daar in vinden?

Reply:
Als webservers last hebben van hoofdletters, mag je dan ook geen hoofdletters in de waarden zetten? Dus zoeken met postcode=1234AA is fout?
api-66 stelt "Use UPPER_SNAKE_CASE for enumeration values", maar zoeken op een query parameter met enumeratiewaarde moet dan de enumeratiewaarde vertalen naar lower case?

Reply:
De naming conventie stelt geen eisen aan waardes van query parameters (alleen keys). Alleen voor enumeraties wordt een standaard beschreven voor waardes, maar dat zullen we op basis van de feedback gaan heroverwegen.